File: src/main/java/com/example/finsentinel/service/trading/uta/BrokerCapability.java
package com.example.finsentinel.service.trading.uta;

/**
 * Capabilities a broker can declare support for.
 * Used by BrokerRegistry to route operations to the correct broker.
 */
public enum BrokerCapability {
    SPOT_TRADING,
    MARGIN_TRADING,
    PERPETUAL_SWAP,
    FUTURES,
    OPTIONS,
    MARKET_DATA,
    ORDER_MANAGEMENT,
    SHORT_SELLING
}

File: packages/shared/src/enums/broker-capability.ts
export const BrokerCapability = {
  SPOT_TRADING: 'SPOT_TRADING',
  MARGIN_TRADING: 'MARGIN_TRADING',
  PERPETUAL_SWAP: 'PERPETUAL_SWAP',
  FUTURES: 'FUTURES',
  OPTIONS: 'OPTIONS',
  MARKET_DATA: 'MARKET_DATA',
  ORDER_MANAGEMENT: 'ORDER_MANAGEMENT',
  SHORT_SELLING: 'SHORT_SELLING',
} as const;

export type BrokerCapability =
  (typeof BrokerCapability)[keyof typeof BrokerCapability];
